lib/app.rb

```diff
@@ -0,0 +1,99 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+#require 'byebug'
+require 'tilt/erb'
+require 'data_mapper'
+require 'dm-migrations'
+require './lib/peep'
+require './lib/users'
+
+class App < Sinatra::Base
+
+  set :views, proc { File.join(root, '..', 'views') }
+  enable :sessions
+  set :session_secret, '123321123'
+  use Rack::Session::Pool
+  env = ENV['RACK_ENV'] || "development"
+  DataMapper.setup(:default, ENV['DATABASE_URL'] || "postgres://localhost/chitter_web_#{env}")
+
+  DataMapper.finalize
+  DataMapper.auto_upgrade!
+
+  register Sinatra::Flash
+
+  get '/' do
+    @peeps = Peep.all(order: [:created_at.desc])
+    erb :index
+  end
+
+  get '/join' do
+    erb :join
+  end
+
+  post '/join' do
+    if ((params[:name] == '') || (params[:email] == '') || (params[:user_name] == '') || (params[:password] == '') || (params[:password_confirm] == ''))
```

Comment:
Metrics/LineLength: Line is too long. [152/90]<br>Style/ParenthesesAroundCondition: Don't use parentheses around the condition of an `if`.